src/compiler/error-detector.js

```diff
@@ -96,3 +94,27 @@ function checkExpression (exp: string, text: string, errors: Array<string>) {
     }
   }
 }
+
+function isValidExpression (exp: string) {
+  try {
+    new Function(`return ${exp}`)
+    return true
+  } catch (e) {
+    /* istanbul ignore next */
+    try {
+      const babylon = (new Function(`return require('babylon')`))()
+      /* $flow-disable-line */
+      babylon.parseExpression(exp, {
+        plugins: [
+          'objectRestSpread',
+          'functionBind',
+          'numericSeparator',
+          'optionalChaining'
+        ]
```

Comment:
I haven't used `babylon` before, so these may not be the most intelligent questions - I apologize in advance. 😅  

- Where do these plugins come from? Are we relying on them being installed by the user? What if one of them isn't?
- Would it be possible to read from a user's `.babelrc` file, if it exists, then simply skip this step if there is none? I worry about having two separate combinations of Babel features, one of which it may _seem_ the user is in control of, but really isn't.